libsql/src/sync.rs

```diff
@@ -31,11 +45,21 @@ impl SyncContext {
             frame_no,
             frame_no + 1
         );
-        let max_frame_no = self
+
+        tracing::debug!("pushing frame");
+
+        let durable_frame_num = self
             .push_with_retry(uri, frame.to_vec(), self.max_retries)
             .await?;
 
-        Ok(max_frame_no)
+        tracing::debug!(?durable_frame_num, "frame successfully pushed");
+
+        // Update our last known max_frame_no from the server.
+        self.durable_frame_num = durable_frame_num;
+
+        self.write_metadata().await?;
```

Comment:
Do we have to write metadata on every frame? Maybe we could do that after each batch?

Reply:
https://github.com/tursodatabase/libsql/pull/1830/commits/de24d8211e4dd42833f962de0188087f896206f1

Reply:
What does this mean?

Reply:
I don't see this being fixed in commit de24d82

Reply:
sorry I made a mistake it wasn't done in that commit but its done here https://github.com/tursodatabase/libsql/pull/1830/commits/26ac07e35e549eb8dfd77ccfde9960de85079800